src/compiler/types/generate-method-types.ts

```diff
@@ -1,13 +1,20 @@
 import type * as d from '../../declarations';
 import { getTextDocs } from '@utils';
 
-export const generateMethodTypes = (cmpMethods: d.ComponentCompilerMethod[]): d.TypeInfo => {
-  return cmpMethods.map((cmpMethod) => ({
-    name: cmpMethod.name,
-    type: cmpMethod.complexType.signature,
-    optional: false,
-    required: false,
-    internal: cmpMethod.internal,
-    jsdoc: getTextDocs(cmpMethod.docs),
-  }));
+/**
+ * Generates the individual event types for all @Method() decorated events in a component
+ * @param cmpMeta component runtime metadata for a single component
+ * @returns the generated type metadata
+ */
+export const generateMethodTypes = (cmpMeta: d.ComponentCompilerMeta): d.TypeInfo => {
+  return [
+    ...cmpMeta.methods.map((cmpMethod) => ({
```

Comment:
Is there a benefit to using the spread operator here? The `map()` method already [creates a new array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

Reply:
Good point - removed in https://github.com/ionic-team/stencil/pull/3316/commits/1f6dc411506f09e1544ffc16eec0e67c8ee6147f